### PR TITLE
feat(plugin-sql): add offset parameter to getMemories for database-le…

### PR DESCRIPTION
**Pull request by 0xbbjoker**
> [!NOTE]
> Adds an optional offset to getMemories for pagination, applies limit/offset in SQL, validates non-negative values, and adds integration tests covering paging and edge cases.
> 
> - **Core**:
>   - Add `offset?: number` to `getMemories` in `packages/core/src/database.ts` and `packages/core/src/types/database.ts`.
> - **Plugin SQL**:
>   - Implement `offset` handling in `BaseDrizzleAdapter.getMemories` with validation (`offset >= 0`).
>   - Apply conditional `.limit()` and `.offset()` to the base query; keep ordering by `createdAt` desc.
>   - Update JSDoc to document `offset`.
> - **Tests**:
>   - Add integration tests for pagination using `count` + `offset`, offset-only, large offset, zero/absent offset behavior, negative offset rejection, and consistency with `countMemories`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 6961feff4ddcef83c12168fa4730e50afe731105. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

## Summary by CodeRabbit

- New Features
  - Added optional offset-based pagination to memory retrieval for more precise paging.

- Tests
  - Introduced comprehensive integration tests covering pagination (count/offset combinations), large-offset edge cases, non-overlapping pages, and partial update integrity.

- Chores
  - Added a new plugin submodule for knowledge functionality.
  - Updated the knowledge plugin submodule reference.

- Documentation
  - Updated parameter docs to include the new offset option in memory retrieval.

### CODE REVIEW

**Comment by 0xbbjoker** [comment]
@coderabbitai summary

File: .gitmodules [hunk 1]
```diff
@@ -1,3 +1,6 @@
 [submodule ".cursor"]
 	path = .cursor
 	url = https://github.com/elizaOS/.cursor
+[submodule "packages/plugin-knowledge"]
```
Comment (wtfsayo) [inline .gitmodules]: has submodule

**Review by wtfsayo — CHANGES_REQUESTED** [review]
